vc-worker/src/main/java/com/vmware/flowgate/vcworker/scheduler/job/VCDataService.java

```diff
@@ -168,11 +190,488 @@ public void checkAndUpdateIntegrationStatus(SDDCSoftwareConfig vc, String messag
+                     logger.error("Format host info map error", e);
+                     return;
+                  }
+                  restClient.saveAssets(hostMappingAsset);
+               } else {
+                  logger.info("No update required");
+                  return;
+               }
+            }
+         }
+      } catch (ConnectionException e1) {
+         checkAndUpdateIntegrationStatus(vc, e1.getMessage());
          return;
+      } catch (ExecutionException e2) {
+         if (e2.getCause() instanceof InvalidLogin) {
+            logger.error("Failed to push data to " + vc.getServerURL(), e2);
+            checkAndUpdateIntegrationStatus(vc, "Invalid username or password.");
+            return;
+         }
+      } catch (Exception e) {
+         logger.error("Failed to sync the host metadata to VC ", e);
+         return;
+      }
+   }
+
+   public boolean feedClusterMetaData(HashMap<String, ClusterComputeResource> clusterMap,
+         HostSystem host, HostInfo hostInfo, String vcInstanceUUID) {
+
+      ManagedObjectReference hostParent = host.getParent();
+      if (hostParent == null || !hostParent.getType().equals("ClusterComputeResource")) {
+         return false;
+      }
+      String clusterMobId = hostParent.getValue();
+      ClusterComputeResource cluster = clusterMap.get(clusterMobId);
+      boolean needUpdate = false;
+      EsxiMetadata oldEsxiMetadata = hostInfo.getEsxiMetadata();
+      if (oldEsxiMetadata == null) {
+         needUpdate = true;
+         oldEsxiMetadata = new EsxiMetadata();
+      }
+      String hostMobId = host._getRef().getValue();
+
+      ConfigInfoEx configInfoExtension = (ConfigInfoEx) cluster.getConfigurationEx();
+
+      String oldEsxiMetadataClusterName = oldEsxiMetadata.getClusterName();
+      if (oldEsxiMetadataClusterName == null || !cluster.getName().equals(oldEsxiMetadataClusterName)) {
+         needUpdate = true;
+         oldEsxiMetadata.setClusterName(cluster.getName());
+      }
+
+      if (!configInfoExtension.getDpmConfigInfo().getEnabled().equals(oldEsxiMetadata.isClusterDPMenabled())) {
+         needUpdate = true;
+         oldEsxiMetadata.setClusterDPMenabled(configInfoExtension.getDpmConfigInfo().getEnabled());
+      }
+
+      if (!configInfoExtension.getDpmConfigInfo().getEnabled().equals(oldEsxiMetadata.isClusterDPMenabled())) {
+         needUpdate = true;
+         oldEsxiMetadata.setClusterDPMenabled(configInfoExtension.getDpmConfigInfo().getEnabled());
+      }
+
+      if (!configInfoExtension.getDpmConfigInfo().getEnabled().equals(oldEsxiMetadata.isClusterDPMenabled())) {
+         needUpdate = true;
+         oldEsxiMetadata.setClusterDPMenabled(configInfoExtension.getDpmConfigInfo().getEnabled());
+      }
+
+      String oldEsxiMetadataClusterDRSBehavior = oldEsxiMetadata.getClusterDRSBehavior();
+      if (oldEsxiMetadataClusterDRSBehavior == null || !configInfoExtension.getDrsConfig().getDefaultVmBehavior().toString()
```

Comment:
double check when drs is off , then what will happen to getDefaultVmBehavior() method.